### DISCUSSION TITLE
Return a single result with Pynetbox.

### DISCUSSION THREAD

**Discussion by AaronJamesMyers** [netbox-community/netbox#9230] [General]
Im new to the Programming world. but ive made some headway here.

I'm trying to write a script that will autoassign the next available interface on a device using pynetbox.

uplinkports = nb.dcim.interfaces.filter(device=UplinkCLLI, connected="false") 

I can iterate and print these results.  But I can't find a way to just extract the first value from the Recordset.

Is there a filter option that will allow me to only return a single results, or a way to select the first returned value.

I can do this using the Netbox API, but was just wondering if i can save a few lines of code and do this with Pynetbox.

**Comment by kkthxbye-code** [comment]
Edit: I misread the post and thought you wanted to create a netbox script.

**Comment by markkuleinio** [comment]
```
>>> list(netbox.ipam.asns.filter())
[AS64800, AS64801, ... ]
```

and

```
>>> next(netbox.ipam.asns.filter())
AS64800
```

or

```
>>> list(netbox.ipam.asns.filter())[0]
AS64800
```

To be clear, the `RecordSet` generator (returned by `.all()` and `.filter()`) can be converted to a list with `list()` and then you can index it normally, like `list(netbox.ipam.asns.filter())[0]` to get the first item (not efficient because it needs to fetch all items first to "blow out" the generator), or you can use the `next()` builtin to take an item from the generator (fast because not all items are fetched).

**Reply by AaronJamesMyers** [reply]
The next() builtin is EXACTLY what I'm looking for. 

Thanks so much.
